yt/utilities/answer_testing/utils.py

```diff
@@ -0,0 +1,491 @@
+    else:
+        return ftrue
+
+#============================================
+# requires_sim
+#============================================
+def requires_sim(sim_fn, sim_type, file_check = False):
+    def ffalse(func):
+        def skip(*args, **kwargs):
+            msg = "{} not found, skipping {}.".format(sim_fn, func.__name__)
+            pytest.fail(msg)
+        return skip
+    def ftrue(func):
+        return func
+    if not can_run_sim(sim_fn, sim_type, file_check):
+        return ffalse
+    else:
+        return ftrue
+
+
+#============================================
+#                create_obj
+#============================================
+def create_obj(ds, obj_type):
+    # obj_type should be tuple of
+    #  ( obj_name, ( args ) )
+    if obj_type is None:
+        return ds.all_data()
+    cls = getattr(ds, obj_type[0])
+    obj = cls(*obj_type[1])
+    return obj
+
+
+#============================================
+#          compare_unit_attributes
+#============================================
+def compare_unit_attributes(ds1, ds2):
+    attrs = ('length_unit', 'mass_unit', 'time_unit',
+             'velocity_unit', 'magnetic_unit')
+    for attr in attrs:
+        u1 = getattr(ds1, attr, None)
+        u2 = getattr(ds2, attr, None)
+        assert u1 == u2
+
+
+#============================================
+#              fake_halo_catalog
+#============================================
+def fake_halo_catalog(data):
+    filename = "catalog.0.h5"
+    ftypes = dict((field, '.') for field in data)
+    extra_attrs = {"data_type": "halo_catalog",
+                   "num_halos": data['particle_mass'].size}
+    ds = {'cosmological_simulation': 1,
+          'omega_lambda': 0.7,
+          'omega_matter': 0.3,
+          'hubble_constant': 0.7,
+          'current_redshift': 0,
+          'current_time': YTQuantity(1, 'yr'),
+          'domain_left_edge': YTArray(np.zeros(3), 'cm'),
+          'domain_right_edge': YTArray(np.ones(3), 'cm')}
+    save_as_dataset(ds, filename, data, field_types=ftypes,
+        extra_attrs=extra_attrs
+    )
+    return filename
+
+
+#============================================
+#                 create_plot
+#+===========================================
+def create_plot(ds, plot_type, plot_field, plot_axis, plot_kwargs = None):
+    # plot_type should be a string
+    # plot_kwargs should be a dict
+    if plot_type is None:
+        raise RuntimeError('Must explicitly request a plot type')
+    cls = getattr(pw, plot_type, None)
+    if cls is None:
+        cls = getattr(particle_plots, plot_type)
+    plot = cls(*(ds, plot_axis, plot_field), **plot_kwargs)
+    return plot
+
+
+#============================================
+#               create_plot2
+#============================================
+def create_plot2(data_source, x_field, y_field, z_field,
+                plot_type, plot_kwargs=None):
+    # plot_type should be a string
+    # plot_kwargs should be a dict
```

Comment:
todo: apply numpy style to this (and make it a docstring)

Reply:
Here's an example @jcoughlin11 

https://yt-project.org/doc/developing/developing.html#docstrings